common/src/main/java/org/opensearch/ml/common/controller/MLRateLimiter.java

```diff
@@ -0,0 +1,141 @@
+
+    private String rateLimitNumber;
+    private TimeUnit rateLimitUnit;
+
+    @Builder(toBuilder = true)
+    public MLRateLimiter(String rateLimitNumber, TimeUnit rateLimitUnit) {
+        this.rateLimitNumber = rateLimitNumber;
+        this.rateLimitUnit = rateLimitUnit;
+    }
+
+    public static MLRateLimiter parse(XContentParser parser) throws IOException {
+        String rateLimitNumber = null;
+        TimeUnit rateLimitUnit = null;
+
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser);
+        while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
+            String fieldName = parser.currentName();
+            parser.nextToken();
+
+            switch (fieldName) {
+                case RATE_LIMIT_NUMBER_FIELD:
+                    rateLimitNumber = parser.text();
+                    break;
+                case RATE_LIMIT_UNIT_FIELD:
+                    rateLimitUnit = TimeUnit.valueOf(parser.text());
+                    break;
+                default:
+                    parser.skipChildren();
+                    break;
+            }
+        }
+        return new MLRateLimiter(rateLimitNumber, rateLimitUnit);
+    }
+
+    public MLRateLimiter(StreamInput in) throws IOException{
+        this.rateLimitNumber = in.readOptionalString();
+        if (in.readBoolean()) {
+            this.rateLimitUnit = in.readEnum(TimeUnit.class);
+        }
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeOptionalString(rateLimitNumber);
+        if (rateLimitUnit != null) {
+            out.writeBoolean(true);
+            out.writeEnum(rateLimitUnit);
+        } else {
+            out.writeBoolean(false);
+        }
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params params) throws IOException {
+        builder.startObject();
+        if (rateLimitNumber != null) {
+            builder.field(RATE_LIMIT_NUMBER_FIELD, rateLimitNumber);
+        }
+        if (rateLimitUnit != null) {
+            builder.field(RATE_LIMIT_UNIT_FIELD, rateLimitUnit);
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    public void update(MLRateLimiter updateContent) {
+        if (updateContent.getRateLimitNumber() != null) {
+            this.rateLimitNumber = updateContent.getRateLimitNumber();
+        }
+        if (updateContent.getRateLimitUnit() != null) {
+            this.rateLimitUnit = updateContent.getRateLimitUnit();
+        }
+    }
+
+    public static MLRateLimiter update(MLRateLimiter rateLimiter, MLRateLimiter updateContent) {
+        if (rateLimiter == null) {
+            return updateContent;
+        } else {
+            rateLimiter.update(updateContent);
+            return rateLimiter;
+        }
+    }
+
+    public static boolean isUpdatable(MLRateLimiter rateLimiter, MLRateLimiter updateContent) {
```

Comment:
let's add comment for these methods specifying in which cases we need this

Reply:
Sure